### PR TITLE
Improves keyword density check by not relying on wordboundary

### LINKED ISSUE #203: The wordboundary regex is broken with consecutive keywords
The wordboundary regex we use isn't working when keywords are used consecutively in a sentence. 
This issue was also present in 2.3.5

The phrase "a keyword and a keyword" will match keyword twice (as it should)
When using "a keyword keyword keyword" will match the keyword only twice, in stead of 3 times. 

This is due to the fact the regex matches the whitespaces around the keyword as well. This screenshot taken from regexr shows it how it is matched. 
screen shot 2015-12-15 at 15 13 44

Only the first and third occurence of the keyword is matched, not the second one.

### PR DESCRIPTION
**Pull request by terw-dan**
Fixes #203

Simply changing the wordboundary regex would give unpredictable results with empty keywords and spaces. 
This PR implements a check where we match per word, like we already did for the passive voice analysis. This makes sure a keyword is matched if it is followed directly by another keyword. 

For testing: Use a sentence with the keyword 2 or more times after eachother. Make sure you use 100+ words to trigger the keyword density check.

### CODE REVIEW

File: js/researches/getKeywordDensity.js [hunk 1]
```diff
@@ -17,6 +21,10 @@ module.exports = function( paper ) {
...
 		return 0;
 	}
-	var keywordCount = matchWords( text, keyword, locale );
-	return ( keywordCount / wordCount ) * 100;
+	var words = getWords( text );
+	var keywordCount = filter( words, function( word ) {
+		return ( word === keyword || transliterate( word, locale ) === keyword );
+	} );
+
+	return ( keywordCount.length / wordCount ) * 100;
```
Comment (jcomack) [inline js/researches/getKeywordDensity.js]: Shouldn't this result be properly formatted before returning it?

**Comment by jcomack** [comment]
CR done 🚧

**Comment by boblinthorst** [comment]
Acceptance: It is indeed found three times in code, but the assessment still says 2, the assessment is incorrect.

**Comment by jcomack** [comment]
CR 2 done 👍

**Comment by jcomack** [comment]
Acceptance done 👍

**Pull request merged**